### PR TITLE
Trace fixes

### LINKED ISSUE #3034: Early wake from DSP when running hotword pipeline with FW debug enabled
When we open wov stream and then try soix, there is early wake from DSP

see attached logs:
https://github.com/thesofproject/linux/files/6783689/soix.txt

Resume caused by IRQ 179, AudioDSP

But when wov stream is closed, the soix entry and exit is fine.

In kernel config,i had enabled FW debug and other options for debug purpose: (everything under SOF_DEBUG)
+CONFIG_SND_SOC_SOF_DEBUG=y
+CONFIG_SND_SOC_SOF_DEBUG_ENABLE_DEBUGFS_CACHE=y
+CONFIG_SND_SOC_SOF_DEBUG_ENABLE_FIRMWARE_TRACE=y
+CONFIG_SND_SOC_SOF_DEBUG_FORCE_IPC_POSITION=y
+CONFIG_SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST=y
+CONFIG_SND_SOC_SOF_DEBUG_RETAIN_DSP_CONTEXT=y
+CONFIG_SND_SOC_SOF_DEBUG_VERBOSE_IPC=y
+CONFIG_SND_SOC_SOF_DEBUG_XRUN_STOP=y
+CONFIG_SND_SOC_SOF_DEVELOPER_SUPPORT=y

### PR DESCRIPTION
**Pull request by ranj063**
Fixes https://github.com/thesofproject/sof/issues/4558

But it should also fix https://github.com/thesofproject/sof/issues/4779 and possibly even https://github.com/thesofproject/linux/issues/3034

And if it does fix #3034, it is replacement for https://github.com/thesofproject/linux/pull/3099

### CODE REVIEW

File: sound/soc/sof/trace.c [hunk 1]
```diff
@@ -556,7 +556,6 @@ void snd_sof_release_trace(struct snd_sof_dev *sdev)
 
 	sdev->dtrace_is_enabled = false;
 	sdev->dtrace_draining = true;
-	wake_up(&sdev->trace_sleep);
 }
```
Comment (kv2019i) [inline sound/soc/sof/trace.c:559]: Hmm, how can you be sure the user-space thread is not in sof_wait_trace_avail() ? The release trace can be called when sof-logger is still running and waiting for traces.

File: sound/soc/sof/intel/hda-dai.c [hunk 2]
```diff
@@ -279,7 +279,7 @@ static int hda_link_pcm_prepare(struct snd_pcm_substream *substream,
 	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
```
Comment (kv2019i) [inline sound/soc/sof/intel/hda-dai.c:279]: typo in commit: dma_sata in subject

**Review by kv2019i — CHANGES_REQUESTED** [review]
Partial review, commenting on the first patch.

**Comment by plbossart** [comment]
@ranj063 I think there should be two separate PRs here
a) your last commit that should apply to all platforms
b) S0ix-related changes - on which I agree with @kv2019i that the fix is controversial.

I don't think we should bundle the two parts, it's better for validation if we split, no?

**Comment by ranj063** [comment]
to be replaced with a better fix

**Pull request closed without merging**